src/sorting/radix_sort.rs

```diff
@@ -0,0 +1,62 @@
+/// Sorts the elements of `arr` in-place using radix sort.
+///
+/// Time complexity is `O((n + b) * logb(k))`, where `n` is the number of elements,
+/// `b` is the base (the radix), and `k` is the largest element.
+/// When `n` and `b` are roughly the same maginitude, this algorithm runs in linear time.
+///
+/// Space complexity is `O(n + b)`.
+pub fn radix_sort(arr: &mut [u64]) {
+    let max: usize = match arr.iter().max() {
+        Some(&x) => x as usize,
+        None => return,
+    };
+    // Make radix a power of 2 close to arr.len() for optimal runtime
+    let radix = arr.len().next_power_of_two();
```

Comment:
How does this help for optimal runtime?

Reply:
It's a simple way to choose a radix of the same magnitude as `n`, which is necessary for the algorithm to run in linear time.